src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MarkMembersAsStatic.Fixer.cs

```diff
@@ -12,19 +22,185 @@ namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+                new MarkMembersAsStaticAction(
+                    MicrosoftQualityGuidelinesAnalyzersResources.MarkMembersAsStaticCodeFix,
+                    ct => MakeStaticAsync(context.Document, root, node, ct)),
+                context.Diagnostics);
+        }
+
+        private async Task<Solution> MakeStaticAsync(Document document, SyntaxNode root, SyntaxNode node, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // Update definition to add static modifier.
+            var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
+            var madeStatic = syntaxGenerator.WithModifiers(node, DeclarationModifiers.Static);
+            document = document.WithSyntaxRoot(root.ReplaceNode(node, madeStatic));
+            var solution = document.Project.Solution;
+
+            // Update references, if any.
+            root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            node = root.DescendantNodes().Single(n => n.SpanStart == node.SpanStart && n.Span.Length == madeStatic.Span.Length);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var symbol = semanticModel.GetDeclaredSymbol(node, cancellationToken);
+            if (symbol != null)
+            {
+                solution = await UpdateReferencesAsync(symbol, solution, cancellationToken).ConfigureAwait(false);
+            }
+
+            return solution;
+        }
+
+        private async Task<Solution> UpdateReferencesAsync(ISymbol symbol, Solution solution, CancellationToken cancellationToken)
         {
-            // Fixer not yet implemented.
-            return Task.CompletedTask;
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var references = await SymbolFinder.FindReferencesAsync(symbol, solution, cancellationToken).ConfigureAwait(false);
+            if (references?.Count() != 1)
+            {
+                return solution;
+            }
+
+            foreach (var referenceLocationGroup in references.Single().Locations.GroupBy(r => r.Document))
+            {
+                // Get document in current solution
+                var document = solution.GetDocument(referenceLocationGroup.Key.Id);
+                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+                // Compute replacements
+                var editor = new SyntaxEditor(root, solution.Workspace);
+                foreach (var referenceLocation in referenceLocationGroup)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    var referenceNode = root.FindNode(referenceLocation.Location.SourceSpan, getInnermostNodeForTie: true);
```

Comment:
I think you need to update root in every iteration in case multiple references are in same file

Reply:
ah. actually no. this code is right, and you don't need nested update since it can't have nested reference.

Reply:
Yes, because it is doing edits based on syntax editor